File: functions/src/models/SplashConfig.ts
export type SplashAnimation = 'fade' | 'scale' | 'slide' | 'bounce' | 'rotate';

export interface SplashConfig {
  enabled: boolean;
  logoUrl: string;
  backgroundColor: string;
  animation: SplashAnimation;
  duration: number;
  showProgressBar: boolean;
  progressBarColor?: string;
  loadingText?: string;
  loadingTextColor?: string;
  showAppVersion: boolean;
  minimumDisplayTime: number;
  fadeOutDuration: number;
}

export const DEFAULT_SPLASH_CONFIG: SplashConfig = {
  enabled: true,
  logoUrl: '',
  backgroundColor: '#1E293B',
  animation: 'fade',
  duration: 2000,
  showProgressBar: true,
  progressBarColor: '#673AB7',
  loadingText: 'Carregando...',
  loadingTextColor: '#FFFFFF',
  showAppVersion: true,
  minimumDisplayTime: 1500,
  fadeOutDuration: 500,
};

export function validateSplashConfig(config: SplashConfig): { valid: boolean; errors: string[] } {
  const errors: string[] = [];
  if (!config.logoUrl) errors.push('logoUrl é obrigatório');
  if (config.duration < 500 || config.duration > 10000) errors.push('duration deve estar entre 500 e 10000ms');
  if (config.minimumDisplayTime < 0 || config.minimumDisplayTime > config.duration) {
    errors.push('minimumDisplayTime deve ser menor que duration');
  }
  return { valid: errors.length === 0, errors };
}

File: codigos antigos/Layout 02/lib/models/splash_config.dart
class SplashConfig {
  final bool enabled;
  final String logoUrl;
  final String backgroundColor;
  final String animation;
  final int duration;
  final bool showProgressBar;
  final String? progressBarColor;
  final String? loadingText;
  final String? loadingTextColor;
  final bool showAppVersion;
  final int minimumDisplayTime;
  final int fadeOutDuration;

  const SplashConfig({
    required this.enabled,
    required this.logoUrl,
    required this.backgroundColor,
    required this.animation,
    required this.duration,
    required this.showProgressBar,
    this.progressBarColor,
    this.loadingText,
    this.loadingTextColor,
    required this.showAppVersion,
    required this.minimumDisplayTime,
    required this.fadeOutDuration,
  });

  factory SplashConfig.fromJson(Map<String, dynamic> json) {
    return SplashConfig(
      enabled: json['enabled'] ?? true,
      logoUrl: json['logoUrl'] ?? '',
      backgroundColor: json['backgroundColor'] ?? '#1E293B',
      animation: json['animation'] ?? 'fade',
      duration: json['duration'] ?? 2000,
      showProgressBar: json['showProgressBar'] ?? true,
      progressBarColor: json['progressBarColor'],
      loadingText: json['loadingText'],
      loadingTextColor: json['loadingTextColor'],
      showAppVersion: json['showAppVersion'] ?? true,
      minimumDisplayTime: json['minimumDisplayTime'] ?? 1500,
      fadeOutDuration: json['fadeOutDuration'] ?? 500,
    );
  }

  static const SplashConfig defaultConfig = SplashConfig(
    enabled: true,
    logoUrl: '',
    backgroundColor: '#1E293B',
    animation: 'fade',
    duration: 2000,
    showProgressBar: true,
    progressBarColor: '#673AB7',
    loadingText: 'Carregando...',
    loadingTextColor: '#FFFFFF',
    showAppVersion: true,
    minimumDisplayTime: 1500,
    fadeOutDuration: 500,
  );
}
